test/regex_test.cpp

```diff
@@ -0,0 +1,186 @@
+#include "gtest/gtest.h"
+#include "regex_yaml.h"
+#include "stream.h"
+
+using YAML::RegEx;
+using YAML::Stream;
+
+namespace {
+TEST(RegExTest, Empty) {
+  RegEx empty;
+  EXPECT_TRUE(empty.Matches(std::string()));
+  EXPECT_EQ(0, empty.Match(std::string()));
+  for (int ii = Stream::eof() + 1; ii < 128; ++ii) {
+    EXPECT_FALSE(empty.Matches(char(ii)));
```

Comment:
Rather than casting to char, what do you think about making the loop variable a char? You won't be able to have < as the terminus of the loop, but maybe this way you could make, e.g., a constexpr vector<char> at the top of this file that consists of all of the chars that you're trying to test, and then just loop through that.